### ISSUE TITLE
Problem about gradient accumulation

### ISSUE THREAD

**Issue by Cwj1212** [NVlabs/stylegan2-ada-pytorch#233]
I observed that the code seems to use gradient accumulation.but 'loss.backward' isn't included in 'with module.no_sync():',only the forward is included in 'with module.no_sync():', If not for gradient accumulation, what is the purpose of  'with module.no_sync():' here,

https://github.com/NVlabs/stylegan2-ada-pytorch/blob/6f160b3d22b8b178ebe533a50d4d5e63aedba21d/training/loss.py#L38-L48
forward process('self.G_mapping(z, c)' and 'self.G_synthesis(ws)') is in the with 'misc.ddp_sync(self.G_mapping, sync):'

https://github.com/NVlabs/stylegan2-ada-pytorch/blob/6f160b3d22b8b178ebe533a50d4d5e63aedba21d/training/loss.py#L73-L74
while backward process is not in 'misc.ddp_sync():'

Since I don't know much about ddp, I'm not quite sure what the purpose of  'misc.ddp_sync():' is here, is it for gradient accumulation, but backward process seems to do a useless gradient synchronization during gradient accumulation
